external/fv3fit/fv3fit/emulation/layers/architecture.py

```diff
@@ -178,3 +261,193 @@ def NoWeightSharingSLP(
+    """
+
+    def __init__(
+        self,
+        feature_lengths: Mapping[str, int],
+        share_conv_weights=False,
+        *args,
+        **kwargs,
+    ):
+        """
+        Args:
+            feature_lengths: Map of output variable names to expected
+                feature dimension length
+        """
+        super().__init__(*args, **kwargs)
+        self._feature_lengths = feature_lengths
+        self._single_dense = share_conv_weights
+
+        if share_conv_weights:
+            layer_cls = tf.keras.layers.Conv1D
+        else:
+            layer_cls = tf.keras.layers.LocallyConnected1D
+
+        self.output_layers = {
+            name: layer_cls(1, 1, name=f"rnn_output_{name}")
+            for name in self._feature_lengths.keys()
+        }
+
+    def call(self, rnn_outputs: tf.Tensor) -> Mapping[str, tf.Tensor]:
+        """
+        Args:
+            hidden_output: Network output from hidden layers of RNN with dim
+                [nsamples, nlev, channels]
+
+        Returns:
+            Connected model outputs by variable
+        """
+
+        fields = {}
+        for name, feature_length in self._feature_lengths.items():
+            if feature_length == 1:
+                rnn_out = rnn_outputs[..., 0:1, :]
+            else:
+                rnn_out = rnn_outputs
+
+            field_out = self.output_layers[name](rnn_out)
+            field_out = tf.squeeze(field_out, axis=-1)
+            fields[name] = field_out
+
+        return fields
+
+
+def _get_output_layer(key, feature_lengths):
+    if key == "rnn-v1":
+        return RNNOutput(feature_lengths)
+    elif key == "rnn-v1-shared-weights":
+        return RNNOutput(feature_lengths, share_conv_weights=True)
+    else:
+        return StandardOutput(feature_lengths)
+
+
+def _get_combine_layer(key):
+    if "rnn" in key:
+        return CombineInputs(combine_axis=-1, expand_axis=-1)
+    else:
+        return CombineInputs(combine_axis=-1, expand_axis=None)
+
+
+def _get_arch_layer(key, kwargs):
+
+    if key == "rnn-v1" or key == "rnn-v1-shared-weights":
+        return RNN(**kwargs)
+    elif key == "rnn":
+        return HybridRNN(**kwargs)
+    elif key == "dense":
+        return MLPBlock(**kwargs)
+    elif key == "linear":
+        if kwargs:
+            raise TypeError("No keyword arguments accepted for linear model")
+        return MLPBlock(depth=0)
+    else:
+        raise KeyError(f"Unrecognized architecture provided: {key}")
+
+
+class _HiddenArchitecture(tf.keras.layers.Layer):
+    def __init__(
+        self,
+        key: str,
+        net_kwargs: Mapping[str, Any],
+        feature_lengths: Mapping[str, int],
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.arch = _get_arch_layer(key, net_kwargs)
+        self.input_combiner = _get_combine_layer(key)
+        self.outputs = _get_output_layer(key, feature_lengths)
+
+    def call(self, tensors: Sequence[tf.Tensor]):
```

Comment:
Output type? I think it is
```suggestion
    def call(self, tensors: Sequence[tf.Tensor]) -> Mapping[str, tf.Tensor]:
```